astrology.lic

```diff
@@ -35,6 +41,20 @@ class Astrology
     result.compact.select { |data| data['circle'] <= DRStats.circle }
   end
 
+  def check_astral
+    return unless DRStats.circle > 99
+    return unless @ap_training
+    return unless @ap_source
+    return unless @ap_destination
+
+    if UserVars.astral_plane_exp_timer.nil? || Time.now - UserVars.astral_plane_exp_timer > 3600
+      wait_for_script_to_complete('bescort', ['ways', @ap_destination])
+      wait_for_script_to_complete('bescort', ['ways', @ap_source])
+      UserVars.astral_plane_exp_timer = Time.now
+      walk_to @safe_room
```

Comment:
My initial thought is that this shouldn't be astrology's concern. I would expect it to be the caller's responsibility to decide where to go next, which may not be the safe_room (crossing-training has a fair bit of logic just for this case, to avoid running around when not necessary).

Are any of the start or end rooms truly dangerous? E.g. populated with mobs?